File: backend/Modeles/compte_model.py
# /backend/models/compte_model.py
from flask_sqlalchemy import SQLAlchemy

db = SQLAlchemy()

class Compte(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    email = db.Column(db.String(255), unique=True, nullable=False)
    numTel = db.Column(db.String(255), unique=True, nullable=False)
    motDePasse = db.Column(db.String(255), nullable=False)
    categorie = db.Column(db.Enum('admin', 'patient', 'docteur', 'clinique', 'laboratoire'), nullable=False)
    confirmation_token = db.Column(db.String(55), nullable=False)
   

    

File: untitled11/lib/models/compte_model.dart
// /frontend/lib/models/compte_model.dart
class Compte {
  String email;
  String numTel;
  String motDePasse;
  String categorie;

  Compte({
    required this.email,
    required this.numTel,
    required this.motDePasse,
    required this.categorie,
  });

  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'numTel': numTel,
      'motDePasse': motDePasse,
      'categorie': "patient",
    };
  }
}
